search/elastic.py

```diff
@@ -579,35 +632,34 @@ def search(self,
         if elastic_filters:
             filter_segment = {
                 "bool": {
-                    "must": elastic_filters
+                    "should": elastic_filters
```

Comment:
Tests helped me to get right search behaviour.
```
{'bool': {'filter': {'bool': {'should': [{'range': {'enrollment_end': {'gte': '2020-07-28T09:44:35.219985'}}},
                                         {'bool': {'must_not': {'exists': {'field': 'enrollment_end'}}}}]}},
          'must': {'bool': {'must': [{'range': {'enrollment_start': {'lte': '2020-07-28T09:44:35.215909'}}}]}}}}

```

[Filter query doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.8/query-dsl-bool-query.html). By filter we strict search documents. By `should` we talk that field should be matched but or it can be empty, or field may not exists.
I changed  `must` to `should` during rewrote query expression to the new ES syntax, so my target is to pass all tests and to keep the existing search behaviour.

Reply:
got it thanks